File: api-gateway-java-reactive/src/main/java/ru/kuramshindev/apigatewayjavareactive/GatewayConfiguration.java
package ru.kuramshindev.apigatewayjavareactive;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.reactive.function.client.WebClient;

@Configuration
public class GatewayConfiguration {

    @Value("${upstreams.base-uri}")
    private String upstreamsBaseUri;

    @Bean
    public WebClient getWebClient(WebClient.Builder webClientBuilder) {
        return webClientBuilder.baseUrl(upstreamsBaseUri).build();
    }

}

File: api-gateway-kotlin-reactive/src/main/kotlin/ru/kuramshindev/apigatewaykotlinreactive/GatewayConfiguration.kt
package ru.kuramshindev.apigatewaykotlinreactive

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.reactive.function.client.WebClient

@Configuration
class GatewayConfiguration(
    @Value("\${upstreams.base-uri}")
    private val upstreamsBaseUri: String
) {

    @Bean
    fun getWebClient(webClientBuilder: WebClient.Builder): WebClient =
        webClientBuilder.baseUrl(upstreamsBaseUri).build()
}